ZitiUpdateService/utils/Settings.cs

```diff
@@ -116,14 +118,37 @@ private void OnCreated(object sender, FileSystemEventArgs e) {
         }
 
         private void OnChanged(object sender, FileSystemEventArgs e) {
-            Logger.Info("Settings file changed. Reloading...");
-            this.Load();
-            this.OnConfigurationChange?.Invoke(null, null);
+            try {
+                this.Load();
+            } catch (IOException ioe) {
+                int isLocked = (int)(System.Runtime.InteropServices.Marshal.GetHRForException(ioe) & 0xFFFF);
+                if (isLocked == 32) {
+                    /*
+                     see https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-
+                     ERROR_SHARING_VIOLATION
+                     32 (0x20)
+                     The process cannot access the file because it is being used by another process.
+                    */
+                    Thread.Sleep(500); //wait 500ms for the file to finish being written
+                    try {
+                        Load();
+                        //this.OnConfigurationChange?.Invoke(null, null);
+                    } catch (Exception ex2) {
+                        Logger.Debug("unexpected error loading settings twice. not trying again. {0}", ex2);
+                    }
+                } else {
+                    Logger.Debug("unexpected error loading settings. file was null? file doesn't exist or file was garbage? {0}", ioe);
+                }
+                this.OnConfigurationChange?.Invoke(null, null);

```

Comment:
is my understanding correct that this only happens when there was an io exception when loading?

Reply:
oh man nice catch. thanks. that was supposed to be inside the previous catch